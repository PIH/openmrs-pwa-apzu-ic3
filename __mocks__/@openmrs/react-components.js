import types from 'redux-types';
// mock methods

const visitRest = {

  createVisit: (params) => {

    const visit = params.visit;

    if (visit.patient != 'patient_uuid' || visit.visitType != 'visit_type_uuid' || visit.location != 'location_uuid') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const encounterRest = {

  createEncounter: (params) => {

    if (params.encounter.encounterType === 'invalid_encounter_type') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const BASIC_TYPES = [
  'FETCH_SUCCEEDED',
  'FETCH_PENDING',
  'SET_SUCCEEDED',
  'SET_REQUESTED',
  'SET_FAILED',
  'FETCH_FAILED',
  'FETCH_REQUESTED',
  'SUCCEEDED',
  'PENDING',
  'FAILED',
  'REQUESTED'
];

const LOGIN_TYPES = {
  LOGIN: types('login', BASIC_TYPES),
  LOGIN_LOCATIONS: types('login_locations', BASIC_TYPES),
  LOGOUT: types('logout', BASIC_TYPES)
}

const SESSION_TYPES = types('session', BASIC_TYPES)


export {
  visitRest,
  encounterRest,
  LOGIN_TYPES,
  SESSION_TYPES
};
