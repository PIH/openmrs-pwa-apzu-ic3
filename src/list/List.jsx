import React from 'react';
import { push } from 'connected-react-router';
import { Label } from 'react-bootstrap';
import { DataGrid } from '@openmrs/react-components';

class List extends React.Component {

  componentDidMount() {

    if (this.fetchListAction() !== undefined) {
      this.props.dispatch(this.fetchListAction());
      this.interval = setInterval(() =>
        this.props.dispatch(this.fetchListAction()), 10000);
    }

    if (this.fetchOtherActions() !== undefined) {
      this.fetchOtherActions().forEach((action) => this.props.dispatch(action));
    }

  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  columnDefs() {
    return [
      { headerName: 'uuid', hide: true, field: 'uuid' },
      { headerName: 'Given Name', field: 'name.givenName' },
      { headerName: 'Family Name', field: 'name.familyName' },
      { headerName: 'Gender', field: 'gender' },
      { headerName: 'Age', field: 'age' }
    ];
  }

  fetchListAction() {
    return undefined;  // needs to be overwritten in implementing classes
  }

  fetchOtherActions() {
    return undefined;
  }

  redirectToInfoPageActionCreator() {
    return push('/'); // needs to be overwritten in implementing classed
  }

  title() {
    return "List";
  }

  render() {
    return (
      <div>
        <h3><Label>{this.title()}</Label></h3>
        <DataGrid
          columnDefs={this.columnDefs()}
          rowData={this.props.rowData}
          rowSelectedActionCreators={[this.redirectToInfoPageActionCreator]}
        />
      </div>
    );
  }

}

export default List;
