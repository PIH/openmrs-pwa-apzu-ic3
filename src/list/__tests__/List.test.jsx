import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { spy } from 'sinon';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { DataGrid } from '@openmrs/react-components';
import List from '../List';

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

spy(List.prototype, 'componentDidMount');

const list = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <List {... props} />
      </Provider>
    );
  }
  return mountedComponent;
};

describe('Component: List', () => {
  beforeEach(() => {
    // in an actual implementation of a list, these would be mapped in
    props = {
      rowData: [],
      dispatch: () => {}
    };
    store = mockStore({});
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(list())).toMatchSnapshot();
    expect(list().find(DataGrid).length).toBe(1);
    expect(list().find(DataGrid).props().rowSelectedActionCreators.length).toBe(1);
    expect(list().find(DataGrid).props().rowSelectedActionCreators[0].name).toBe("redirectToInfoPageActionCreator");
    expect(list().find(DataGrid).props().rowSelectedActionCreators[0]().payload.args[0]).toBe("/");
    expect(List.prototype.componentDidMount.calledOnce).toBe(true);
  });
});
