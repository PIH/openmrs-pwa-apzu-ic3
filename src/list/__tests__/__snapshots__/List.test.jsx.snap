// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: List renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <List
    dispatch={[Function]}
    rowData={Array []}
  >
    <div>
      <h3>
        <Label
          bsClass="label"
          bsStyle="default"
        >
          <span
            className="label label-default"
          >
            List
          </span>
        </Label>
      </h3>
      <Connect(DataGrid)
        columnDefs={
          Array [
            Object {
              "field": "uuid",
              "headerName": "uuid",
              "hide": true,
            },
            Object {
              "field": "name.givenName",
              "headerName": "Given Name",
            },
            Object {
              "field": "name.familyName",
              "headerName": "Family Name",
            },
            Object {
              "field": "gender",
              "headerName": "Gender",
            },
            Object {
              "field": "age",
              "headerName": "Age",
            },
          ]
        }
        rowData={Array []}
        rowSelectedActionCreators={
          Array [
            [Function],
          ]
        }
      >
        <DataGrid
          columnDefs={
            Array [
              Object {
                "field": "uuid",
                "headerName": "uuid",
                "hide": true,
              },
              Object {
                "field": "name.givenName",
                "headerName": "Given Name",
              },
              Object {
                "field": "name.familyName",
                "headerName": "Family Name",
              },
              Object {
                "field": "gender",
                "headerName": "Gender",
              },
              Object {
                "field": "age",
                "headerName": "Age",
              },
            ]
          }
          dispatch={[Function]}
          rowData={Array []}
          rowSelectedActionCreators={
            Array [
              [Function],
            ]
          }
        >
          <div
            style={
              Object {
                "height": "600px",
                "width": "100%",
              }
            }
          >
            <AgGridReact
              columnDefs={
                Array [
                  Object {
                    "field": "uuid",
                    "headerName": "uuid",
                    "hide": true,
                  },
                  Object {
                    "field": "name.givenName",
                    "headerName": "Given Name",
                  },
                  Object {
                    "field": "name.familyName",
                    "headerName": "Family Name",
                  },
                  Object {
                    "field": "gender",
                    "headerName": "Gender",
                  },
                  Object {
                    "field": "age",
                    "headerName": "Age",
                  },
                ]
              }
              enableSorting={true}
              id="omrsGrid"
              onGridReady={[Function]}
              onSelectionChanged={[Function]}
              rowClassRules="rowClassRules"
              rowData={Array []}
              rowSelection="single"
            >
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            </AgGridReact>
          </div>
        </DataGrid>
      </Connect(DataGrid)>
    </div>
  </List>
</Provider>
`;
